qiskit/transpiler/passes/routing/utils.py

```diff
@@ -0,0 +1,98 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A set of routines used by all routing methods"""
+
+import numpy as np
+import scipy.sparse as sp
+import scipy.sparse.csgraph as cs
+
+
+def weighted_distance(num_qubits, coupling_map, properties,
+                      alpha_dis=0.5, alpha_error=0.5, alpha_time=0):
+    """Distance between qubits weighted by 2Q gate errors, 2Q SWAP shortest
+       path and 2Q execution time.
+
+    Parameters:
+        num_qubits (int): Number of qubits.
+        coupling_map (CouplingMap): Coupling map.
+        properties (BackendProperties): A backend properties instance.
+        alpha_dis (float): Weight of swap distance matrix.
+        alpha_error (float): Weight of swap error rate matrix.
+        alpha_time (float): Weight of swap execution time matrix.
+
+    Additional Information:
+        math:
+            The error rate of a swap gate
+                E(SWAP.q_i,SWAP.q_j) = 1 - F(q_i,q_j) * F(q_j,q_i) * max(F(q_i,q_j),F(q_j,q_i))
+                F(q_i,q_j) = 1 - E(q_i,q_j)
+            The execution time of a swap gate
+                T(SWAP.q_i,SWAP.q_j) = T(q_i,q_j) + T(q_j,q_i) + min(T(q_i,q_j),T(q_j,q_i))
+            The weighted distance matrix
+                D = alpha_dis * swap_distance_matrix +
+                    alpha_error * swap_error_rate_matrix +
+                    alpha_time * swap_execution_time_matrix
+
+    Returns:
+        ndarray: 2D array of distances.
```

Comment:
When returning Numpy arrays it's good to describe the shape of the array as well.